### ISSUE TITLE
eos_snapshot renders yaml and json outputs incorrectly.

### ISSUE THREAD

**Issue by friday963** [aristanetworks/ansible-avd#2394]
### Issue Summary

I've been testing the snapshot capability and noticed that the snapshots for both json and yaml continuously dump the same output to file.  Whatever the first command is in your list of commands ("commands_list") just keeps being iterated over.  If you used a loop.index in the jinja template you should be able to keep track of your loops and dump each command separately.  
For example: https://stackoverflow.com/questions/12145434/how-to-output-loop-counter-in-python-jinja-template

Here is the problem line in both templates:
{{ hostvars[node].cli_output.results[0].stdout[0] | to_nice_yaml(indent=2) | trim | indent(6) }} <--- You keep iterating over results[0] instead of the correct command.

The templates here: 
https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_snapshot/templates/fabric_json_output.j2 

and here: 
https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_snapshot/templates/fabric_yaml_output.j2

### Which component(s) of AVD impacted

eos_snapshot

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
site.yml

<truncated>
-name: set default vars
 set_fact:
   output_format:
    - yaml
    - json
   commands_list:
    - show lldp neighbors
    - show ip interface brief
    - show interfaces description
- name: collect commands
  import_role:
    name: eos_snapshot
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by carlbuchmann** [comment]
@friday963 Thanks for reporting this issue; additionally to what you pointed out, I ensured we capture the show command with JSON output so that it's helpful! This fix will be included in 3.8.1.

**Comment by carlbuchmann** [comment]
@friday963 AVD 3.8.1 has been released, feedback is welcome!
